do-not-commit/default-mutable-dict.py

```diff
@@ -0,0 +1,393 @@
+import copy
+
+# ruleid: default-mutable-dict
+def assign_func1(default={}):
+    default["potato"] = 5
+
+
+# ruleid: default-mutable-dict
+def assign_func2(default={}):
```

Comment:
Function assign_func2 mutates default dict default. Instances of default function arguments in Python are shared between function calls.  Mutating a default function argument changes the result for all function consumers. This will most likely cause unexpected results, or lead to security vulnerabilities whereby one function consumer can either view or modify the data of another function consumer. ([.common-mistakes.default-mutable-dict.default-mutable-dict](https://semgrep.dev/r/.common-mistakes.default-mutable-dict.default-mutable-dict))
#
<div>
<a href="https://semgrep.dev">
<img align="right" src="https://r2c.dev/semgrep_icon_text.svg" height=28px alt="" />
</a>

<sub> [How do I suppress this?](https://semgrep.dev/docs/running-rules/#managing-findings) **·** [Report a bug](https://github.com/returntocorp/semgrep-rules/issues/new?assignees=&labels=&template=rule_bug_report.md&title=.common-mistakes.default-mutable-dict.default-mutable-dict) **·** [Manage rule](https://semgrep.dev/manage/projects)</sub>
</div>